packages/core-data/src/types/page.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * Internal dependencies
+ */
+import {
+	CommentStatus,
+	Context,
+	ContextualField,
+	PingStatus,
+	PostStatus,
+	RawField,
+	EditedRecord,
+	WithoutNevers,
+} from './common';
+
+interface FullPage< C extends Context > {
+	/**
+	 * The date the post was published, in the site's timezone.
+	 */
+	date: string | null;
+	/**
+	 * The date the post was published, as GMT.
+	 */
+	date_gmt: ContextualField< string | null, 'view' | 'edit', C >;
+	/**
+	 * The globally unique identifier for the post.
+	 */
+	guid: ContextualField< RawField< C >, 'view' | 'edit', C >;
+	/**
+	 * Unique identifier for the post.
+	 */
+	id: number;
+	/**
+	 * URL to the post.
+	 */
+	link: string;
+	/**
+	 * The date the post was last modified, in the site's timezone.
+	 */
+	modified: ContextualField< string, 'view' | 'edit', C >;
+	/**
+	 * The date the post was last modified, as GMT.
+	 */
+	modified_gmt: ContextualField< string, 'view' | 'edit', C >;
+	/**
+	 * An alphanumeric identifier for the post unique to its type.
+	 */
+	slug: string;
+	/**
+	 * A named status for the post.
+	 */
+	status: ContextualField< PostStatus, 'view' | 'edit', C >;
+	/**
+	 * Type of post.
+	 */
+	type: string;
+	/**
+	 * A password to protect access to the content and excerpt.
+	 */
+	password: ContextualField< string, 'edit', C >;
+	/**
+	 * Permalink template for the post.
+	 */
+	permalink_template: ContextualField< string, 'edit', C >;
+	/**
+	 * Slug automatically generated from the post title.
+	 */
+	generated_slug: ContextualField< string, 'edit', C >;
+	/**
+	 * The ID for the parent of the post.
+	 */
+	parent: ContextualField< number, 'view' | 'edit', C >;
+	/**
+	 * The title for the post.
+	 */
+	title: RawField< C >;
+	/**
+	 * The content for the post.
+	 */
+	content: ContextualField<
+		{
+			/**
+			 * Data as it exists in the database.
+			 */
+			raw: ContextualField< string, 'edit', C >;
+			/**
+			 * Data transformed for display.
+			 */
+			rendered: string;
+			/**
+			 * Whether the content is protected with a password.
+			 */
+			is_protected: boolean;
+			/**
+			 * Version of the content block format used by the page.
+			 */
+			block_version: ContextualField< string, 'edit', C >;
+		},
+		'view' | 'edit',
+		C
+	>;
```

Comment:
is this one not using the `RawField` wrapper because the permissions on each field don't match the pattern?

Reply:
further, when we collapse this to `string` for editing does that imply we cannot change the `block_version` for it?

Reply:
> is this one not using the RawField wrapper because the permissions on each field don't match the pattern?

I just didn't like how the `&` read, but now that we're using RawField in the `Updatable` type wrapper, it makes sense to actually do it.

> further, when we collapse this to string for editing does that imply we cannot change the block_version for it?

I believe so, yes. It is even marked as `"readonly": true` in the API docs.